src/api/app/controllers/source_controller.rb

```diff
@@ -336,82 +334,6 @@ def delete_project_pubkey
     end
   end
 
-  def require_package_name
-    required_parameters :project, :package
-
-    @project_name = params[:project]
-    @package_name = params[:package]
-
-    valid_package_name! @package_name
-  end
-
-  # GET /source/:project/:package/_meta
-  def show_package_meta
-    require_package_name
-
-    pack = Package.get_by_project_and_name(@project_name, @package_name, use_source: false)
-
-    if params.key?(:meta) || params.key?(:rev) || params.key?(:view) || pack.nil?
-      # check if this comes from a remote project, also true for _project package
-      # or if meta is specified we need to fetch the meta from the backend
-      path = request.path_info
-      path += build_query_from_hash(params, [:meta, :rev, :view])
-      pass_to_backend path
-      return
-    end
-
-    render xml: pack.to_axml
-  end
-
-  # PUT /source/:project/:package/_meta
-  def update_package_meta
-    require_package_name
-
-    rdata = Xmlhash.parse(request.raw_post)
-
-    if rdata['project'] && rdata['project'] != @project_name
```

Comment:
@coolo do you think that's validation still valid? I was trying to submit a meta to a different project and package than the pointed by the URL and it doesn't woork..

Reply:
not sure what your question is. If it's "should it be possible to submit any meta to any project?" than no, while super annoying, it can save your ass if you do scripted updates to _meta and have a typo in the filename. The backend will refuse the update anyway, so possibly you don't need that code - if *that* was the question :)

Reply:
yes, that was my question.. if the url has a resource but the package and
project are pointing to something different, if i would update the
"something different".